Libraries/LibWeb/HTML/HTMLLinkElement.cpp

```diff
@@ -596,8 +598,31 @@ void HTMLLinkElement::resource_did_load_favicon()
     document().check_favicon_after_loading_link_resource();
 }
 
-static NonnullRefPtr<Core::Promise<Web::Platform::DecodedImage>> decode_favicon(ReadonlyBytes favicon_data, URL::URL const& favicon_url, GC::Ref<DOM::Document> document)
+static NonnullRefPtr<Core::Promise<bool>> decode_favicon(ReadonlyBytes favicon_data, URL::URL const& favicon_url, GC::Ref<DOM::Document> document)
```

Comment:
nit: you should possibly get away with `Core::Promise<void>` here, as you seem not to use this `bool` anywhere?